gm4_smelteries/data/gm4_smelteries/functions/process.mcfunction

```diff
@@ -0,0 +1,14 @@
+# smeltery processes
+# @s - smeltery
+# called by main
+
+# removal
+execute unless block ~ ~ ~ hopper run function gm4_smelteries:destroy
+
+# verify furnace location
+execute unless block ^ ^ ^1 furnace run function gm4_smelteries:find_furnace
```

Comment:
Bug : Due to the way these conditions are designed, the smeltery can "lose" its furnace if it's broken and replaced at the same location. This is due to it's `gm4_bf_has_furnace` tag being removed, but there still being a furnace at ^ ^ ^1 that prevents it from regaining that tag.